test/Tests/src/Data/Range_Spec.enso

```diff
@@ -70,6 +72,50 @@ spec = Test.group "Range" <|
         0.up_to 1 . not_empty . should_be_true
         0.up_to 5 . not_empty . should_be_true
         5.down_to 0 . not_empty . should_be_true
+    Test.specify "should allow getting by index using at" <|
```

Comment:
There should be a generic test suite working on `Vector`, `List`, `Array` (as soon as @hubertp aligns its methods) making sure the behavior is the same on all implementation of these _"vector-like"_ objects.